### PR TITLE
Handle umbrella logfiles elegantly in makeflow

### PR DESCRIPTION
**Pull request by hmeng-19**
Add umbrella logfiles into makeflow log file
Add umbrella-log-prefix opt
Fix umbrella wrapper

### CODE REVIEW

**Comment by hmeng-19** [comment]
This PR aims to fix #1468 .
@nhazekam , PTAL.

**Comment by hmeng-19** [comment]
@nhazekam , for testing this feature, cctools/makeflow/example/makeflow_umbrella will be a good example.

File: makeflow/src/dag_node.c [hunk 1]
```diff
@@ -47,6 +47,7 @@ struct dag_node *dag_node_create(struct dag *d, int linenum)
 	n->resource_request = CATEGORY_ALLOCATION_FIRST;
 
 	n->umbrella_spec = NULL;
+	n->umbrella_logfile = NULL;
```
Comment (nhazekam) [inline makeflow/src/dag_node.c]: With umbrella_spec we were planning (at some point) on the specification being unique to the node, are we also expecting umbrella_logfile to be unique to the node. If not this should only exist in the wrapper so the file names are consistent across all of the nodes.

File: makeflow/src/makeflow.c [hunk 2]
```diff
@@ -149,6 +149,8 @@ static char *mountfile = NULL;
 static char *mount_cache = NULL;
 static int use_mountfile = 0;
 
+static char *umbrella_log_prefix = NULL;
```
Comment (nhazekam) [inline makeflow/src/makeflow.c]: This prefix belongs in the umbrella wrapper, not in makeflow.c. Refer to how we handle logs for the resource monitor.

File: makeflow/src/dag_node.c (hunk 1 above)
Comment (hmeng-19) [inline makeflow/src/dag_node.c]: @nhazekam , good point. 
I do not think it is very meaningful to allow the user to specify an unique umbrella log file for each rule.
I will move this into the umbrella_wrapper.

File: makeflow/src/makeflow.c (hunk 2 above)
Comment (hmeng-19) [inline makeflow/src/makeflow.c]: @nhazekam , sounds good. Will fix it.

**Comment by hmeng-19** [comment]
@nhazekam , PTAL.
https://github.com/cooperative-computing-lab/cctools/pull/1491/commits/7bbf0c81eb9f5ae8568d20eb9dda0c0f1e2c72f0 should fix your concerns.

**Comment by hmeng-19** [comment]
@nhazekam , PTAL.
As we discussed earlier, I added a new commit, https://github.com/cooperative-computing-lab/cctools/pull/1491/commits/e983a9aa842e7238d748e4ecb0817f6ac0ed7f83, to implement the `--umbrella-mode` option, which allows the user to specify the execution mode of umbrella at makeflow level (not rule level).

**Comment by hmeng-19** [comment]
@nhazekam , PTAL. I rebased it and fixed the conflicts.

**Pull request merged**